### PR TITLE
fix(sdl): use the new draw buffer structure in the layers

### LINKED ISSUE #5533: LV_USE_DRAW_SDL currently broken
### LVGL version

v9.0.0-65-ge4de2d265

### What happened?

When I enable `LV_USE_DRAW_SDL`, compilation fails as follows:

```
[  6%] Building C object lvgl/CMakeFiles/lvgl.dir/src/draw/sdl/lv_draw_sdl.c.o
/.../lvgl/src/draw/sdl/lv_draw_sdl.c: In function ‘draw_to_texture’:
/.../lvgl/src/draw/sdl/lv_draw_sdl.c:177:15: error: ‘lv_layer_t’ {aka ‘struct _lv_layer_t’} has no member named ‘buf’
  177 |     dest_layer.buf = lv_draw_buf_align(sdl_render_buf, LV_COLOR_FORMAT_ARGB8888);
      |               ^
make[2]: *** [lvgl/CMakeFiles/lvgl.dir/build.make:888: lvgl/CMakeFiles/lvgl.dir/src/draw/sdl/lv_draw_sdl.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:132: lvgl/CMakeFiles/lvgl.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

This seems to be due to an API change, but I don't know how to fix it.

### How to reproduce?

Build with `LV_USE_DRAW_SDL`.

### PR DESCRIPTION
**Pull request by kisvegabor**
### Description of the feature or fix

fixes #5533

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html)
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by kisvegabor** [comment]
A few notes:
- The SDL render is not finished yet, it's more like a proof of concept which can be improved if needed
- `libsdl2-image-dev` needs to installed and `-lSDL2_image` needs to be linked
- You can use the images like this: `lv_image_set_src(img1, "path/to/image.png")`
- In the future we should use LVGL's built in image decoders
- Triangle, vector, and box shadow draw tasks are not implemented yet
- In the texture cache the texts are compared by address, so dynamically allocated texts are cached multiple times.
- For simple draw tasks, e.g. screen fill we shouldn't use a texture but use SDL's fill directly.

**Review by C47D — APPROVED** [review]
LGTM

**Comment by C47D** [comment]
> Triangle, vector, and box shadow draw tasks are not implemented yet

Should we open a new issue to keep track of this missing feature?

**Review by FASTSHIFT — DISMISSED** [review]

**Review by PGNetHun — DISMISSED** [review]

File: src/draw/sdl/lv_draw_sdl.c [hunk 1]
```diff
@@ -172,13 +172,16 @@ static bool draw_to_texture(lv_draw_sdl_unit_t * u, cache_data_t * data)
 {
     lv_draw_task_t * task = u->task_act;
 
+    lv_area_t a;
+    _lv_area_intersect(&a, u->base_unit.clip_area, &task->area);
```
Comment (XuNeo) [inline src/draw/sdl/lv_draw_sdl.c]: Is the result `a` not used?

Comment (kisvegabor) [reply src/draw/sdl/lv_draw_sdl.c]: True, removed

**Comment by kisvegabor** [comment]
> Should we open a new issue to keep track of this missing feature?

Yes, I'll do that.

**Review by PGNetHun — APPROVED** [review]

**Comment by kisvegabor** [comment]
The CI should be fixed by #5597

**Pull request merged**